File: leet_easy_c++/leet_206/leet_206a.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* iter = head;
        stack<int> stk;
        while(iter){
            stk.emplace(iter->val);
            iter = iter->next;
        }
        iter = head;
        while(iter){
            iter->val = stk.top();
            stk.pop();
            iter = iter->next;
        }
        return head;
    }
};

File: leet_easy_javascript/leet_206/leet_206a.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    return rLh(null, head);
};

/**
 * @param {ListNode} prev
 * @param {ListNode} cur
 * @return {ListNode}
 */
var rLh = function(prev, cur) {
    if(cur === null) return prev;
    var next = cur.next;
    cur.next = prev;
    return rLh(cur, next);
};